File: calculator-java/library/src/main/java/com/example/calculator/service/IVisitor.java
package com.example.calculator.service;

import com.example.calculator.service.nodes.BinNode;
import com.example.calculator.service.nodes.NumberNode;
import com.example.calculator.service.nodes.UnaryNode;

public interface IVisitor {

    Number visit(NumberNode numberNode);

    Number visit(BinNode binNode);

    Number visit(UnaryNode unaryNode);

}

File: calculator-csharp/CalculatorLib/CalculatorService/IVisitor.cs
using CalculatorService.Nodes;


namespace CalculatorService
{
    /// <summary>Interface for visitor design pattern to visit nodes</summary>
    public interface IVisitor
    {
        Number visit(NumberNode numberNode);
        Number visit(BinNode binNode);
        Number visit(UnaryNode unaryNode);
    }
}
